src/otx/core/model/entity/classification.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Class definition for classification model entity used in OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Union
+
+from mmengine.runner import load_checkpoint
+
+# classification.model.backbones should be initialized to register the backbones.
+import otx.algo.classification.model.backbones  # noqa: F401
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.classification import MulticlassClsBatchDataEntity, MulticlassClsBatchPredEntity
+from otx.core.model.entity.base import OTXModel
+from otx.core.utils.config import convert_conf_to_mmconfig_dict
+
+if TYPE_CHECKING:
+    from mmpretrain.models.utils import ClsDataPreprocessor
+    from omegaconf import DictConfig
+    from torch import nn
+
+
+class OTXClassificationModel(OTXModel[MulticlassClsBatchDataEntity, MulticlassClsBatchPredEntity]):
+    """Base class for the classification models used in OTX."""
+
+class MMPretrainCompatibleModel(OTXClassificationModel):
+    """Classification model compatible for MMPretrain.
+
+    It can consume MMPretrain model configuration translated into OTX configuration
+    (please see otx.tools.translate_mmrecipe) and create the OTX classification model
+    compatible for OTX pipelines.
+    """
+
+    def __init__(self, config: DictConfig) -> None:
+        self.config = config
+        self.load_from = config.pop("load_from", None)
+        super().__init__()
+
+    def _create_model(self) -> nn.Module:
+        from mmpretrain.registry import MODELS
+        ### DataPreprocessor config should be converted as tuple
+        ### If not, it MODELS.build() creates the empty data preprocessor
+        ### That's the reason why I added the lines below
+        data_preprocessor_cfg = self.config.pop("data_preprocessor")
+        converted_data_preprocessor_cfg = convert_conf_to_mmconfig_dict(
+            data_preprocessor_cfg, to="list")
+        try:
+            converted_cfg = convert_conf_to_mmconfig_dict(self.config, to="tuple")
+            converted_cfg.data_preprocessor = converted_data_preprocessor_cfg.to_dict()
+            model = MODELS.build(converted_cfg)
+        except AssertionError:
+            converted_cfg = convert_conf_to_mmconfig_dict(self.config, to="list")
+            converted_cfg.data_preprocessor = converted_data_preprocessor_cfg.to_dict()
+            model = MODELS.build(converted_cfg)
+
+        if self.load_from is not None:
+            load_checkpoint(model, self.load_from)
```

Comment:
This can be wrapped up with other `mm*`.

Reply:
I guess that `MODELS.build(cfg)` and `load_checkpoint()` is common for all `mm*`

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2678/commits/80ebb392bf2ca7e67f4b1806c9549de3062e3eed